backoffice/src/Services/DroitService.cs

```diff
@@ -254,6 +259,38 @@ public async Task<Droit> GetDroitByReferenceAsync(string reference)
     }
 
 
+    public async Task UploadWmCsvForm(List<WMRowDto> wreckMaterials,Guid droitId)
+    {
+
+        foreach (var wmForm in wreckMaterials.Select(rowDto => new WreckMaterialForm
```

Comment:
Maybe some error handling here. Also some unit tests passing in a hard coded (WMRowDto) and getting the expected outcomes. => be easier to split out the Parsing to another function to unit test => 


foreach (var wmForm in wreckMaterials.Select(rowDto => createWMFormFromCsvRow(rowDto))

Reply:
Maybe an autoMapper WMRowDto => WreckMaterialForm